File: 234-palindrome-linked-list/234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        
        sol_list=[]
        
        while(head!=None):
            sol_list.append(head.val)
            head=head.next

        i=0
        j=len(sol_list)-1
        
        while(i<j):
            
            if (sol_list[i]!=sol_list[j]):
                return False
            i+=1
            j-=1
            
        return True

File: 234-palindrome-linked-list/234-palindrome-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

int List_len(struct ListNode* head){
    
    int len=0;
    
    while(head){
        
        head=head->next;
        len++;
    }
    return len;
}


bool isPalindrome(struct ListNode* head){
    
    int Length;
    
    Length=List_len(head);
    
    int arr[Length];
    
    for (int i=0;i<Length;i++){
        arr[i]=head->val;
        head=head->next;
    }
    
    int j=0;
    int k=Length-1;
    
    while(j<k){
        
        if(arr[j]!=arr[k])
            return false;
        j++;
        k--;
    }
    
    return true;
    
}
